app/components/form/fields/TlsCertsField.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useState } from 'react'
+import type { Control } from 'react-hook-form'
+import { useController } from 'react-hook-form'
+
+import { Button, Error16Icon, FieldLabel, MiniTable, Modal } from '@oxide/ui'
+
+import { DescriptionField, FileField, NameField } from 'app/components/form'
+import type { SiloCreateInput } from 'app/forms/silo-create'
+import { useForm } from 'app/hooks'
+import { readBlobAsText } from 'app/util/file'
+
+export function TlsCertsField({ control }: { control: Control<SiloCreateInput> }) {
+  const [showAddCert, setShowAddCert] = useState(false)
+
+  const {
+    field: { value: items, onChange },
+  } = useController({ control, name: 'tlsCertificates' })
+
+  return (
+    <>
+      <div className="max-w-lg">
+        <FieldLabel id="tls-certificates-label" className="mb-3">
+          TLS Certificates
+        </FieldLabel>
+        {!!items.length && (
+          <MiniTable.Table className="mb-4">
+            <MiniTable.Header>
+              <MiniTable.HeadCell>Name</MiniTable.HeadCell>
+              {/* For remove button */}
+              <MiniTable.HeadCell className="w-12" />
+            </MiniTable.Header>
+            <MiniTable.Body>
+              {items.map((item, index) => (
+                <MiniTable.Row
+                  tabIndex={0}
+                  aria-rowindex={index + 1}
+                  aria-label={`Name: ${item.name}, Description: ${item.description}`}
+                  key={item.name}
```

Comment:
Currently we aren't checking for uniqueness around name. Do we want to do so? If not, we should use something like `${item.name}-${index}` to ensure the keys do not clash.

Reply:
Yeah, we probably should enforce uniqueness. I guess we’d check that on submit in the modal? Refuse to submit if there’s a duplicate.